lib/ReactViews/Map/Panels/SharePanel/SharePanel.tsx

```diff
@@ -0,0 +1,667 @@
+        <div className={DropdownStyles.section}>
+          <div>{t("share.printTitle")}</div>
+          <div className={Styles.explanation}>
+            {t("share.printExplanation")}
+          </div>
+          <div>
+            <button
+              className={Styles.printButton}
+              disabled={this.state.isDownloading}
+              onClick={() => {
+                this.setState({
+                  isDownloading: true
+                });
+                this.props.terria.currentViewer
+                  .captureScreenshot()
+                  .then(dataString => {
+                    downloadImg(dataString);
+                  })
+                  .finally(() =>
+                    this.setState({
+                      isDownloading: false
+                    })
+                  );
+              }}
+            >
+              {t("share.downloadMap")}
+            </button>
+            <button
+              className={Styles.printButton}
+              onClick={() => {
+                const newWindow = window.open();
+                this.props.viewState.setPrintWindow(newWindow);
+              }}
+            >
+              {t("share.printViewButton")}
+            </button>
+          </div>
+        </div>
+        <hr className={Styles.thinLineDivider} />
+        <div className={classNames(DropdownStyles.section, Styles.shortenUrl)}>
+          <div className={Styles.btnWrapper}>
+            <button
+              type="button"
+              onClick={this.toggleAdvancedOptions}
+              className={Styles.btnAdvanced}
+            >
+              <span>{t("share.btnAdvanced")}</span>
+              {this.advancedIsOpen() ? (
+                <Icon glyph={Icon.GLYPHS.opened} />
+              ) : (
+                <Icon glyph={Icon.GLYPHS.closed} />
+              )}
+            </button>
+          </div>
+          {this.advancedIsOpen() && (
+            <>
+              <div className={DropdownStyles.section}>
+                <p className={Styles.paragraph}>{t("share.embedTitle")}</p>
+                <Input
+                  large
+                  dark
+                  type="text"
+                  readOnly
+                  placeholder={this.state.placeholder}
+                  value={iframeCode}
+                  onClick={e => {
+                    const target = e.target as HTMLInputElement;
+                    return target.select();
+                  }}
+                />
+              </div>
+              {this.isUrlShortenable() && (
+                <div className={Styles.shortenUrl}>
+                  <Checkbox
+                    textProps={{ small: true }}
+                    id="shortenUrl"
+                    isChecked={(this.shouldShorten() as boolean) ?? false}
+                    onChange={this.onShortenClicked}
+                    className={Styles.checkbox}
+                  ></Checkbox>
+                  <p>{t("share.shortenUsingService")}</p>
```

Comment:
Checkbox label should go as a child of Checkbox component so it is accessible and label is clickable, and the padding is currently a bit off (this change should fix it), check https://github.com/TerriaJS/terriajs/blob/5ad78fd8fccf78877982dc8a41ec39ff78750c01/lib/ReactViews/Map/Panels/SettingPanel.tsx#L277-L289 for sample implementation

Also, we should use TextSpan here instead of `<p>`